edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -98,7 +98,8 @@ def authenticate_credentials(self, payload):
         if username is None:
             raise exceptions.AuthenticationFailed('JWT must include a preferred_username or username claim!')
         try:
-            user, __ = get_user_model().objects.get_or_create(username=username)
+            user, created = get_user_model().objects.get_or_create(username=username)
+            logger.info(f"User {user.id} found for username {username} (created?: {created}")
```

Comment:
Adding this line will inform whether the exception is being thrown on the `get_or_create` call to the `User` model or whether the exception is being thrown on the `user.save()` below.